clients/python/src/tensorzero/client.py

```diff
@@ -38,18 +39,89 @@
 )
 
 
-class AsyncTensorZeroGateway:
+class BaseTensorZeroGateway:
```

Comment:
Use `from abc import ABC, abstractmethod` + `abstractmethod` on interfaces for inference/feedback (if possible with async)